falcon/request.py

```diff
@@ -1028,6 +1059,22 @@ def _parse_form_urlencoded(self):
 
             self._params.update(extra_params)
 
+    def _parse_rfc_forwarded(self):
+        """rfc 7239"""
+        addr = []
+        for forwarded in self.env['HTTP_FORWARDED'].split(','):
+            for param in forwarded.split(';'):
+                param = param.strip().split('=', 1)
+                if len(param) == 1:
+                    continue
+                key, val = param
+                if key.lower() not in ('for', 'by'):
+                    # we only want for/by params
+                    continue
+                host, _ = parse_host(unquote_string(val))
```

Comment:
Practically speaking, do we need to do anything beyond simply removing the enclosing double quotes?

Looking at RFC 7239, quoted strings are only required for IPv6 addresses, since those contain invalid token characters. Obfuscated identifiers and IPv4 nodes do not contain invalid token characters, and so do not need to be quoted (although the sender may choose to do so). Regardless, none of the allowed characters for the different node identifiers need to be escaped (with a backslash), and so it seems unlikely that a sender would ever do so.

Indeed, RFC 7230 states that senders should only use escaping when absolutely necessary. On the other hand, the RFC requires that the receiver unescape any `quoted-pair`'s, if present. But again, it seems highly unlikely that a Falcon app would ever come across any `quoted-pair`'s in the wild within the context of a Forwarded header.

Interestingly enough, Werkzeug doesn't strictly follow RFC 7320; `werkzeug.http.unquote_header_value` [only handles][1] escaped backslashes and double quotes. These are the two characters that RFC 7320 specifically calls out as the only cases in which a sender should generate a `quoted-pair`. But note that these two characters are actually disallowed by the ABNF for the Forwarded header.

[1]: https://github.com/mitsuhiko/werkzeug/blob/master/werkzeug/http.py#L198